src/query.ts

```diff
@@ -22,6 +22,7 @@ export type QueryInput<
 > &
   ExpressionAttributeNames<KeyConditionExpression> &
   ExpressionAttributeNames<FilterExpression> &
+  ExpressionAttributeNames<ProjectionExpression> &
```

Comment:
This will also detect values, eg :val in a projection expression. I had no idea this was valid.

Reply:
Not sure what you mean, that type only detects names it seems and the docs make it seem like it can only contain `#` names.

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ProjectionExpressions.html

Reply:
Oh my memory is wrong. Never mind, didn't think hard enough lol